File: src/test/java/com/rafaelfiume/tictactoe/support/ConsoleInputReaderStub.java
package com.rafaelfiume.tictactoe.support;

import com.rafaelfiume.tictactoe.BoardPosition;
import com.rafaelfiume.tictactoe.console.ConsoleInputReader;

import java.util.Queue;
import java.util.concurrent.ArrayBlockingQueue;

public class ConsoleInputReaderStub implements ConsoleInputReader {

    private final Queue<Integer> positions = new ArrayBlockingQueue(10);

    @Override
    public BoardPosition readUserInput() {
        return BoardPosition.of(positions.poll());
    }

    public void willReturn(int position) {
        positions.add(position);
    }

}

File: src/test/java/com/rafaelfiume/tictactoe/support/ConsoleInputReaderStub.kt
package com.rafaelfiume.tictactoe.support

import com.rafaelfiume.tictactoe.BoardPosition
import com.rafaelfiume.tictactoe.console.ConsoleInputReader
import java.util.*

import java.util.concurrent.ArrayBlockingQueue

class ConsoleInputReaderStub : ConsoleInputReader {

    private val positions: Queue<Int> = ArrayBlockingQueue(10)

    override fun readUserInput() = BoardPosition.of(positions.poll())

    fun willReturn(position: Int) {
        positions.add(position)
    }

}
